pkg/controllers/resources/nodes/translate.go

```diff
@@ -187,39 +190,33 @@ func (s *nodeSyncer) translateUpdateStatus(ctx *synccontext.SyncContext, pNode *
 						// count pods that are not synced by this vcluster
 						nonVClusterPods++
 					}
-					for _, container := range pod.Spec.InitContainers {
-						cpu -= container.Resources.Requests.Cpu().MilliValue()
-						memory -= container.Resources.Requests.Memory().Value()
-						storageEphemeral -= container.Resources.Requests.StorageEphemeral().Value()
-					}
-					for _, container := range pod.Spec.Containers {
-						cpu -= container.Resources.Requests.Cpu().MilliValue()
-						memory -= container.Resources.Requests.Memory().Value()
-						storageEphemeral -= container.Resources.Requests.StorageEphemeral().Value()
+
+					reqs, _ := resourceutil.PodRequestsAndLimits(&pod)
```

Comment:
so it seems that we were summing up init containers instead of picking the one with the biggest resource request + not accounting for `pod.spec.overhead`, right?

Reply:
yes exactly. This lead our calculations to be incorrect as summing them up caused them to go beyond the maximum allocatable size of the node, and yes pod.spec.overhead was never taken into account.